who_l3_smart_tools/core/l2/data_dictionary.py

```diff
@@ -0,0 +1,324 @@
+import os
+from collections import defaultdict
+from typing import Optional
+
+from openpyxl import load_workbook
+
+from who_l3_smart_tools.core.l2.utils import (
+    remove_special_characters,
+    to_camel_case,
+)
+from who_l3_smart_tools.utils.jinja2 import (
+    DATA_TYPE_MAP,
+    initalize_jinja_env,
+    render_to_file,
+)
+
+jinja_env = initalize_jinja_env(__name__)
+
+
+# pylint: disable=too-many-instance-attributes
+class L2Row:
+    """
+    Represents a row in the L2 data dictionary.
+
+    Args:
+        raw_row (dict): The raw row data.
+        coding_data_element (Optional[str]): The coding data element.
+
+    Attributes:
+        raw_row (dict): The raw row data.
+        activity_id (str): The activity ID.
+        data_element_id (Optional[str]): The data element ID.
+        data_element_label (Optional[str]): The data element label.
+        description (Optional[str]): The description and definition.
+        choice_type (Optional[str]): The multiple choice type (if applicable).
+        data_type (Optional[str]): The data type.
+        input_options (Optional[str]): The input options.
+        validation_condition (Optional[str]): The validation condition.
+        required (Optional[str]): The required flag.
+        coding_data_element (Optional[str]): The coding data element.
+
+    Methods:
+        validate_coding_data_element: Validates the coding data element.
+        _activity_id_to_invariant: Converts the activity ID to an invariant.
+        _get_questionare_title: Gets the questionnaire title.
+        to_invariant: Converts the row to an invariant.
+        to_concept_item: Converts the row to a concept item.
+        to_model_item: Converts the row to a model item.
+        to_questionnaire_item: Converts the row to a questionnaire item.
+        to_valueset_item: Converts the row to a valueset item.
+    """
+
+    DEFAULT_INVARIANT_EXPRESSION = "<NOT-IMPLEMENTED>"
+    DEFAULT_INVARIANT_SEVERITY = "error"
+
+    def __init__(
+        self, raw_row: dict, coding_data_element: Optional[str] = None
+    ) -> None:
+        self.raw_row = raw_row
+        self.activity_id = raw_row["Activity ID"]
+        self.data_element_id = raw_row["Data Element ID"]
+        self.data_element_label = raw_row["Data Element Label"]
+        self.description = raw_row["Description and Definition"]
+        self.choice_type = raw_row["Multiple Choice Type (if applicable)"]
+        self.data_type = raw_row["Data Type"]
+        self.input_options = raw_row["Input Options"]
+        self.validation_condition = raw_row["Validation Condition"]
+        self.required = raw_row["Required"]
+        self.coding_data_element = coding_data_element
+
+    @property
+    def map_cardinality(self):
```

Comment:
Since this is a `@property`, it would probably be better-named `cardinality` rather than `map_cardinality`. `L2Row.cardinality` feels more natural than `L2Row.map_cardinality`